app/controllers/courses/accredited_body_controller.rb

```diff
@@ -99,6 +108,14 @@ def build_course_params
       @autocompleted_provider_code = params[:course].delete(:autocompleted_provider_code)
     end
 
+    def build_back_link
+      @back_link_path = if @provider.sites.count > 1
+                          new_provider_recruitment_cycle_courses_locations_path(course: @course_creation_params)
+                        else
+                          new_provider_recruitment_cycle_courses_study_mode_path(course: @course_creation_params)
+                        end
+    end
```

Comment:
This `build_back_link`, and all the other versions of it in the other controllers, is doing my head in. That is, until I have a look at `NextCourseCreationStepService`.

I feel like we should have a single service that handles both `next` and `back` ... I know that breaks the principle of having an `execute` as a single point of entry for services ... maybe we could have `Providers::CourseCreation::NextStepService` and `Providers::CourseCreation::BackService`, with the commonalities kept in `Providers::CourseCreation`? That might be messy if we have to define `NextStepService` and `BackService` in a separate Ruby file, which would be necessary for the auto-loader to work (and Rubocop not to complain, probably).

Not sure what's the best approach, but I do think that this needs a bit of cleanup, it's not clear what the logic is doing here.

Reply:
Agreed, its doing my head as well.

``` ruby
def oscillate(current, course) do
{
  previous: previous_link_factory,
  next: next_link_factory,
  roulette_spin: lucky_day 
}
end

```

Reply:
Yeah I'll make a service first, then I'll look at some common interface for getting both the next and back links :thinking: I did avoid making a service for this reason, it's a bit of a headache to get a ncie way to do it but i'll dedicate some time to it